### PR TITLE
Adds a function to verify cron jobs auth token

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 10 Jan 2024

Developer Name: @Ajeyakrishna-k 

----

## Description

Adds a function to verify cron jobs auth token

**Documentation Updated?**

-   [ ] Yes
-   [x] No

**Breaking Changes**

-   [ ] Yes
-   [x] No

**Development Tested?**

-   [x] Yes
-   [ ] No

**Tested in Staging?**

-   [x] Yes
-   [ ] No

**Under Feature Flag**

-   [x] Yes
-   [ ] No

**Database Changes**

-   [ ] Yes
-   [x] No

### Screenshots

Screenshot 2024-01-10 at 10 57 08 PM

### CODE REVIEW

File: src/utils/verifyAuthToken.ts [hunk 1]
```diff
@@ -24,3 +24,23 @@ export async function verifyAuthToken(authHeader: string, env: env) {
...
 }
+
+/**
+ *
+ * @param authHeader { string } : the auth header of request
+ * @param env { env }: the ctx (context) which contains the secrets put in as wrangler secrets.
+ */
+
+export async function verifyCronJobsToken(authHeader: string, env: env) {
+  const parts = authHeader.split(" ");
```
Comment (iamitprakash) [inline src/utils/verifyAuthToken.ts:35]: what if authHeader is null

File: src/utils/verifyAuthToken.ts [hunk 2]
```diff
@@ -24,3 +24,23 @@ export async function verifyAuthToken(authHeader: string, env: env) {
...
+  if (parts.length !== 2 || parts[0] !== "Bearer") {
+    throw new Error(INVALID_TOKEN_FORMAT);
+  }
+  const authToken = parts[1];
```
Comment (iamitprakash) [inline src/utils/verifyAuthToken.ts:39]: use better variable name

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: src/utils/verifyAuthToken.ts (hunk 1 above)
Comment (Ajeyakrishna-k) [reply src/utils/verifyAuthToken.ts]: Done

File: src/utils/verifyAuthToken.ts (hunk 2 above)
Comment (Ajeyakrishna-k) [reply src/utils/verifyAuthToken.ts]: Done

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**